### PR TITLE
Fix a bug which returned True for ask(Q.imaginary(0**I))

### PR DESCRIPTION
**Pull request by debugger22**
The bug appears in https://github.com/sympy/sympy/pull/9731

### CODE REVIEW

**Comment by asmeurer** [comment]
I think this just masking a bug in the handlers. We should fix it there instead.

**Comment by debugger22** [comment]
The result is being returned from here.

**Comment by asmeurer** [comment]
I think you might as well include this fix with #9731, since you can't even add a test for it without that change.

File: sympy/assumptions/handlers/sets.py [hunk 1]
```diff
@@ -430,6 +430,7 @@ def Symbol(expr, assumptions):
     def _number(expr, assumptions):
         # let as_real_imag() work first since the expression may
         # be simpler to evaluate
+        expr = refine(expr)  # Evaluate unevaluated numbers
         r = expr.as_real_imag()[0].evalf(2)
```
Comment (asmeurer) [inline sympy/assumptions/handlers/sets.py]: I think the bug is here.  In https://github.com/sympy/sympy/pull/9731, `(0**I).as_real_imag()` gives `(re(0**I), im(0**I))`, and `re(0**I).evalf(2)` gives `0`.  In fact, `(0**I).evalf(2)` gives `0`, which is wrong. In fact, this bug even manifests itself in master as `(x**I).evalf(subs={x:0})`.

**Comment by asmeurer** [comment]
I think calling refine in the handlers in general is probably a bad idea. For one, it could slow them down, but more importantly, refine itself calls ask, so the potential for infinite recursion is quite high.

**Comment by debugger22** [comment]
I'm creating a new PR for that.

**Comment by debugger22** [comment]
Here's the issue https://github.com/sympy/sympy/issues/9807

**Pull request closed without merging**